src/backend/joanie/payment/models.py

```diff
@@ -414,33 +483,26 @@ class Meta:
         verbose_name = "credit card"
         verbose_name_plural = "credit cards"
         ordering = ["-created_on"]
-        constraints = [
-            models.UniqueConstraint(
-                condition=models.Q(is_main=True),
-                fields=["owner"],
-                name="unique_main_credit_card_per_user",
-            )
-        ]
+
+    def add_owner(self, owner):
+        """
+        Add new owner to the credit card owners field through the `CreditCardOwnership` model to
+        enforce the validation logic (through the clean() method) for the field `is_main`
+        of ownership each time it's saved.
+        """
+        ownership = CreditCardOwnership(
+            owner=owner,
+            credit_card=self,
+        )
+        ownership.save()
```

Comment:
Just as a shortcut
https://docs.djangoproject.com/en/5.1/ref/models/querysets/#django.db.models.query.QuerySet.create
```suggestion
        CreditCardOwnership.objects.create(
            owner=owner,
            credit_card=self,
        )
```